docs/trouble-shooting.md

```diff
@@ -0,0 +1,60 @@
+# Troubleshooting Guide
+
+Common issue users might have when using Cluster API Provider for Azure.
+
+## Debugging cluster creation
+
+While cluster build out is running, you can optionally follow the controller logs in a separate window as follows:
+
+```bash
+time kubectl get po -o wide --all-namespaces -w # Watch pod creation until azure-provider-controller-manager-0 is available
+
+kubectl logs -n capz-system azure-provider-controller-manager-0 manager -f # Follow the controller logs
+```
+
+An error such as the following in the manager could point to a mismatch between a current CAPI and an old CAPZ version:
+
+```
+E0320 23:33:33.288073       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create AzureMachine VM: failed to create nic capz-cluster-control-plane-7z8ng-nic for machine capz-cluster-control-plane-7z8ng: unable to determine NAT rule for control plane network interface: strconv.Atoi: parsing \"capz-cluster-control-plane-7z8ng\": invalid syntax"  "controller"="azuremachine" "request"={"Namespace":"default","Name":"capz-cluster-control-plane-7z8ng"}
+```
+
+### Debugging workload clusters
+After the workload cluster is finished deploying you will have a kubeconfig in `./kubeconfig`.
+
+Using the ssh information provided during cluster creation (environment variable `AZURE_SSH_PUBLIC_KEY`), you can debug most issues by SSHing into the VMs that have been created:
+
+```
+# connect to first control node - capi is default linux user created by deployment
+apiserver=$(kubectl get azurecluster capz-cluster -o jsonpath='{.status.network.apiServerIp.dnsName}')
+ssh capi@${apiserver}
+
+# list nodes
+kubectl get azuremachines
+NAME                               READY   STATE
+capz-cluster-control-plane-2jprg   true    Succeeded
+capz-cluster-control-plane-ck5wv   true    Succeeded
+capz-cluster-control-plane-w4tv6   true    Succeeded
+capz-cluster-md-0-s52wb            true    Succeeded
+capz-cluster-md-0-w8xxw            true    Succeeded
+
+# pick node name from output above:
+node=$(kubectl get azuremachine capz-cluster-md-0-s52wb -o jsonpath='{.status.addresses[0].address}') 
+ssh -J capi@${apiserver} capi@${node} 
+```
+
+> There are some [provided scripts](/hack/debugging/Readme.md) that can help automate a few common tasks.
+
+Reviewing the following logs on the workload cluster:
+
+- `/var/lib/waagent/custom-script/download/0/stdout`
```

Comment:
Two of these are filenames and two are commands. Just for parallel construction, maybe we should prepend `less` to #1 and #3 so they are also commands?